ord_schema/validations.py

```diff
@@ -326,6 +380,23 @@ def validate_addition_speed(message):
     del message  # Unused.
 
 
+def validate_crude_component(message):
+    if not message.reaction_id:
+        warnings.warn('CrudeComponents must specify a reaction_id',
+                      ValidationError)
+    if (message.HasField('has_derived_amount') and message.has_derived_amount
+            and message.HasField('amount')):
+        warnings.warn(
+            'CrudeComponents with derived amounts cannot have their'
+            ' mass or volume specified explicitly', ValidationError)
+    if ((not message.HasField('has_derived_amount')
+         or message.has_derived_amount is False)
```

Comment:
Prefer `not message.has_derived_amount` to comparing to `False`; you've already covered the unset case.

Reply:
If we use FrozenMessage throughout the validations we can clean a lot of this up...not in this PR though :)